docs/content/preview/contribute/core-database/merge-with-upstream-repositories.md

```diff
@@ -0,0 +1,1017 @@
+   git switch -c add-documentdb-subtree master
+   git subtree add --prefix=src/postgres/third-party-extensions/documentdb documentdb v0.102-0 --no-squash
+   ```
+
+   If `git subtree` does not work, you may need to install it.
+   There may be an alternative method to do this using `git merge -s subtree`, but that has not been tested as of yet.
+   In any case, `git subtree` adds some useful metadata to the commit message, so using that is preferred.
+
+   There should be no merge conflicts since this is a brand new upstream repository in a brand new directory.
+   Any modifications to integrate this upstream repository to YugabyteDB should be done as a followup after this whole process.
+   In this example, this was done by commit [f8e9b50f83a4561b1314057297e3ba3236c7703c](https://github.com/yugabyte/yugabyte-db/commit/f8e9b50f83a4561b1314057297e3ba3236c7703c).
+1. Add the upstream repository details into [`upstream_repositories.csv`][upstream-repositories-csv].
+   Commit this single change as one commit on top of the subtree merge commit.
+   In this example, this was done by commit [3201d6f146c957b39ce23f3c59e76ca31ffa1a0c](https://github.com/yugabyte/yugabyte-db/commit/3201d6f146c957b39ce23f3c59e76ca31ffa1a0c).
+1. Submit this feature branch as reference for review.
+   Phorge is not sufficient since it strips Git metadata.
+   One example of a place to submit this is a personal GitHub fork of [yugabyte/yugabyte-db][repo-yugabyte-db].
+   The reviewer should make sure the subtree merge is done properly and that exactly one commit is added after that concerning [`upstream_repositories.csv`][upstream-repositories-csv].
+1. Create a Phorge revision of this feature branch for official review.
+   Make sure lint is not skipped to make sure [`upstream_repositories.csv`][upstream-repositories-csv] is valid.
+   The reviewer should make sure the latest commit hash of both submissions are exactly equal.
+1. Request permission from @buildteam in the internal slack #eng-infra channel to land a merge commit of this revision.
+1. Land the change using the merge strategy:
+
+   ```sh
+   arc land --strategy merge --revision D...... --onto master
+   ```
+
+   Append `--hold` if it is your first time and you are afraid of making mistakes.
+   After [checking the structure looks good](#subtree-illustrations) using a command such as `git log --oneline --graph`, push the change using the sample command provided at the end of the `arc land --hold` output.
+   In this example, this was done by commit [b01e29bafa6fb8bfb899f0b3ac6e98363340c8e7](https://github.com/yugabyte/yugabyte-db/commit/b01e29bafa6fb8bfb899f0b3ac6e98363340c8e7).
+
+### Converting a squash embedded upstream repository to subtree
+
+Working with an existing squash embedded repositories to the subtree strategy is a lot more complicated.
+This is primarily because of the hoops one has to jump through to make `git blame` point to the subtree rather than the old squash commits.
```

Comment:
```suggestion
This is primarily because of the hoops you have to jump through to make `git blame` point to the subtree rather than the old squash commits.
```